data_types/src/columns.rs

```diff
@@ -379,6 +379,15 @@ impl Deref for ColumnSet {
     }
 }
 
+impl<I> From<I> for ColumnSet
+where
+    I: IntoIterator<Item = i64>,
+{
+    fn from(ids: I) -> Self {
+        Self(ids.into_iter().map(ColumnId::new).collect())
```

Comment:
This is invalid because it breaks Eq and Co. Internally the data is sorted, so please just use the `new` method that does this:

```suggestion
        Self::new(ids.into_iter().map(ColumnId::new).collect())
```